CefSharp.Core/ManagedCefBrowserAdapter.cpp

```diff
@@ -0,0 +1,32 @@
+#include "Stdafx.h"
+#include "Cef.h"
+
+double ManagedCefBrowserAdapter::GetZoomLevel()
+{
+    auto browser = _clientAdapter->GetCefBrowser();
+
+    if (browser != nullptr)
+    {
+        auto host = browser->GetHost();
+        if (CefCurrentlyOn(TID_UI))
+        {
+            return host->GetZoomLevel();
+        }
+        else
+        {
+            // TODO: Add an async version of GetZoomLevel at some point.
+            // NOTE: Use of ManualResetEvent is required here in order
+            // for simple marshaling of some kind of synchronization primitive
+            // to the callback.
+            Task<double>^ task = Cef::UIThreadTaskFactory->StartNew(gcnew Func<double>(this, &ManagedCefBrowserAdapter::GetZoomLevel));
```

Comment:
I haven't had time to read over the `SyncronizationContext` stuff in any details, so I'll be guided by you on this one.

I think we take the pragmatic approach, something that's simple and workable. The `ZoomLevel` property may stay for the `WPF` implementation, just moved to the `IWpfWebBeowser` interface.